allennlp/data/semparse/worlds/nlvr_world.py

```diff
@@ -189,130 +239,218 @@ def execute(self, logical_form: str) -> bool:
+            if len(function_name_parts) == 3:
+                # These are cases like ``object_color_equals``, ``box_count_greater`` etc.
+                attribute_type = function_name_parts[1]
+                comparison_op = function_name_parts[2]
+            elif function_name_parts[2] == 'count':
+                # These are cases like ``object_color_count_equals``,
+                # ``object_shape_count_greater_equals`` etc.
+                attribute_type = "_".join(function_name_parts[1:3])
+                comparison_op = "_".join(function_name_parts[3:])
+            else:
+                # These are cases like ``box_count_greater_equals``, ``object_shape_not_equals``
+                # etc.
+                attribute_type = function_name_parts[1]
+                comparison_op = "_".join(function_name_parts[2:])
+
+        entity_expression = sub_expression[1]
+        returned_count = None
+        returned_attribute = None
+        if entity_type == "box":
+            # You can only count boxes. The other attributes do not apply.
+            returned_count = self.count(self._execute_box_filter(entity_expression))
+        elif "count" in attribute_type:
+            # We're counting objects, colors or shapes.
+            count_function = self._count_functions[attribute_type]
+            returned_count = count_function(self._execute_object_filter(entity_expression))
+        else:
+            # We're getting colors or shapes from objects.
+            attribute_function = self._attribute_functions[attribute_type]
+            returned_attribute = attribute_function(self._execute_object_filter(entity_expression))
+
+        if comparison_op in ["all_equals", "any_equals", "none_equals"]:
+            set_comparison = self._set_binary_operators[comparison_op]
+            if returned_attribute is None:
+                logger.error("Invalid assertion function: %s", sub_expression[0])
+                raise ExecutionError("Invalid assertion function")
+            return set_comparison(returned_attribute, target_attribute)
         else:
-            return self.count(self._execute_object_filter(arguments_list))
+            number_comparison = self._number_operators[comparison_op]
+            if returned_count is None:
+                logger.error("Invalid assertion function: %s", sub_expression[0])
+                raise ExecutionError("Invalid assertion function")
+            return number_comparison(returned_count, target_attribute)
 
     def _execute_box_filter(self, sub_expression: Union[str, List]) -> Set[Box]:
         """
         Box filtering functions either apply a filter on a set of boxes and return the filtered set,
         or return all the boxes.
         The elements should evaluate to one of the following:
-            box_filtering_function(set_to_filter, attribute_function, constant)
-            all_boxes
+        ``(box_filtering_function set_to_filter constant)`` or
```

Comment:
Same comment here - what's stopping you from just having a bunch of methods called `member_count_equals`, etc.?